relay-config/src/config.rs

```diff
@@ -1465,6 +1465,12 @@ impl Config {
         Ok(())
     }
 
+    /// Updates the credentials without creating the file, for testing purposes.
+    pub fn update_credentials(&mut self) {
+        let creds = Credentials::generate();
+        self.credentials = Some(creds);
+    }
```

Comment:
is it possible to conditionally compile such functions only on testing? I tried the cfg(test) trick, but when i run `make test-rust-all` it fails to find the function, despite that it should compile in testing mode. Strangely, running the tests directly from vscode works.

Reply:
I also made some attempts playing around with features and didn't manage. I'm open to updating this if somebody knows a better way.

Reply:
The`#[cfg(test)]` does not propagate to the external crates, and only work within the current one.

It's possible to achieve this, if really really necessary with `features`, by e.g. adding `testing` feature to the crate and hiding all the functionality behind that feature. Then in the crate which has to use this `testing` functionality you will need to add maybe `dev` dependency and enable the feature.